frontend/src/page/profile.tsx

```diff
@@ -47,6 +49,46 @@ export const Profile: React.FC = () => {
         onChange={(e) => {
           setUser({...user, bio: e.target.value});
         }}/>
+      <input
+        accept="image/*"
+        style={{display: 'none'}}
+        id="raised-button-file"
+        multiple
+        type="file"
+        onChange={async (e) => {
+          console.log(e.target.value);
+          const reader = new FileReader();
+          const [file] = e.target.files;
+          reader.onload = async () => {
+            setUserAvatar({...userAvatar, data: reader.result as string});
+
+            const avatarCreatePromise = abortableJsonFetch<StoredUserAvatar>(
+              API_PROFILE_AVATAR_CREATE,
+              Auth.Required,
+              {},
+              {
+                body: userAvatar.data,
```

Comment:
So I can't use the new state immediately here....should I move the avatarCreatePromise into a `useEffect` function? :thinking: